### PR TITLE
statistics: make getIntColumnRowCount correct

### PR DESCRIPTION
**Pull request by winoros**
If we use  `a > 1` to build range in physical plan builder phase, will get `(1, +∞)` first, then we will get a []types.IntColumnRange like `[{2, +∞}]`.
However, this range will call function greaterRowCount(). So the result actually is calculated as `(2, +∞)`.
This pr is to fix this.
More tests will be added in the next pr that introduce selectivity calculation.

PTAL @shenli @hanfei1991 @coocood @lamxTyler

### CODE REVIEW

File: statistics/histogram.go [hunk 1]
```diff
@@ -166,24 +166,27 @@ func (hg *Histogram) equalRowCount(sc *variable.StatementContext, value types.Da
 }
 
 // greaterRowCount estimates the row count where the column greater than value.
```
Comment (shenli) [inline statistics/histogram.go]: Update the comment and add comment for the newly added argument.

File: statistics/histogram.go [hunk 2]
```diff
@@ -166,24 +166,27 @@ func (hg *Histogram) equalRowCount(sc *variable.StatementContext, value types.Da
...
+		gtCount -= eqCount
 	}
-	gtCount := hg.totalRowCount() - lessCount - eqCount
 	if gtCount < 0 {
 		gtCount = 0
 	}
 	return gtCount, nil
 }
 
 // lessRowCount estimates the row count where the column less than value.
```
Comment (shenli) [inline statistics/histogram.go]: So as here.

File: statistics/statistics_test.go [hunk 3]
```diff
@@ -169,7 +169,7 @@ func (s *testStatisticsSuite) TestBuild(c *C) {
 	count, err = col.equalRowCount(sc, types.NewIntDatum(10000))
 	c.Check(err, IsNil)
 	c.Check(int(count), Equals, 1)
-	count, err = col.lessRowCount(sc, types.NewIntDatum(20000))
+	count, err = col.lessRowCount(sc, types.NewIntDatum(20000), false)
```
Comment (shenli) [inline statistics/statistics_test.go]: Any test case covers hasEq is true?

**Comment by hanfei1991** [comment]
It's better to add the count of `equal 2` directly.

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
@lamxTyler PTAL

**Pull request merged**